tools/lsp/properties.rs

```diff
@@ -353,6 +361,178 @@ pub(crate) fn query_properties<'a>(
+                };
+                let edits = vec![lsp_types::OneOf::Left(edit)];
+                let text_document_edits = vec![lsp_types::TextDocumentEdit {
+                    text_document: lsp_types::OptionalVersionedTextDocumentIdentifier::new(
+                        uri.clone(),
+                        version,
+                    ),
+                    edits,
+                }];
+                Some(lsp_types::WorkspaceEdit {
+                    document_changes: Some(lsp_types::DocumentChanges::Edits(text_document_edits)),
+                    ..Default::default()
+                })
+            } else {
+                None
+            }
+        })
+        .flatten()
+}
+
+pub(crate) fn set_binding<'a>(
+    document_cache: &mut DocumentCache,
+    uri: &lsp_types::Url,
+    element: &ElementRc,
+    property_name: &str,
+    new_expression: String,
+) -> Result<SetBindingResponse, Error> {
+    let (mut diag, expression_node) = {
+        let mut diagnostics = BuildDiagnostics::default();
+
+        let syntax_node = i_slint_compiler::parser::parse_expression_as_bindingexpression(
+            &new_expression,
+            &mut diagnostics,
+        );
+
+        let parsed_len: usize = syntax_node
+            .first_child()
+            .expect("A BindingNode needs to contain a child")
+            .text_range()
+            .len()
+            .into();
+        let expression_len = new_expression.len();
+
+        if !diagnostics.has_error() && parsed_len < expression_len {
+            diagnostics.push_error_with_span(
+                format!(
+                    "Expected end of string, found \"{}\"",
+                    &new_expression[parsed_len..expression_len]
+                ),
+                SourceLocation { source_file: None, span: Span::new(parsed_len) },
+            )
+        }
+        (diagnostics, syntax_node)
+    };
+
+    let new_expression_type = {
+        let element = element.borrow();
+        let node = element.node.as_ref().expect("Element had no node!");
+
+        crate::util::with_property_lookup_ctx(document_cache, node, property_name, |ctx| {
+            let expression =
+                i_slint_compiler::expression_tree::Expression::from_binding_expression_node(
+                    expression_node,
+                    ctx,
+                );
+            expression.ty()
+        })
+        .expect("Failed to set up evaluation context.")
+    };
+
+    let property = match get_property_information(
+        element,
+        &mut &mut document_cache.offset_to_position_mapper(uri.clone()),
+        property_name,
+    ) {
+        Ok(p) => Some(p),
+        Err(e) => {
+            diag.push_error_with_span(
+                e.to_string(),
+                i_slint_compiler::diagnostics::SourceLocation {
+                    source_file: None,
+                    span: i_slint_compiler::diagnostics::Span::new(0),
+                },
+            );
+            None
```

Comment:
I suggest to short circuit here and return early (with the result that just has the diagnostics). After all, you can be certain that there is nothing else to do.